packages/client/tests/functional/_utils/setupTestSuiteClient.ts

```diff
@@ -179,5 +180,16 @@ export function setupTestSuiteClientDriverAdapter({
     return { adapter: new PrismaLibSQL(client), __internal }
   }
 
+  if (driverAdapter === AdapterProviders.JS_D1) {
+    const { connectD1 } = require('wrangler-proxy') as typeof import('wrangler-proxy')
+    const { PrismaD1 } = require('@prisma/adapter-d1') as typeof import('@prisma/adapter-d1')
+
+    // Note: default hostname is `http://127.0.0.1:8787`
+    // The custom port is set in packages/client/tests/functional/_utils/wrangler.toml
+    const db = connectD1('MY_DATABASE', { hostname: 'http://127.0.0.1:9090' })
+
+    return { adapter: new PrismaD1(db) }
```

Comment:
How does it work without the wasm engine?

```suggestion
    return { adapter: new PrismaD1(db), __internal }
```

Reply:
Tu run it with the wasm engine and wasm runtime, you can look for the newly added `--client-runtime`

Reply:
Thanks, I don't know anything about this yet